src/main.go

```diff
@@ -91,13 +95,39 @@ func addKeys(akd *keys.SSHAuthorizedKeysDir, name string, force bool) error {
 		k = append(k, b...)
 	}
 
-	// TODO(vc): akd.Add() treats keys as opaque objects, so we need to
-	// sanity check them ourselves before adding them.
-	// update-ssh-keys employs ssh-keygen -l -f for this purpose, can do
-	// same here.
+	if !validKeys(k) {
+		return fmt.Errorf("key(s) invalid.")
+	}
+
 	return akd.Add(name, k, *flagReplace, force)
 }
 
+// validKeys validates a byte slice contains valid ssh keys.
+func validKeys(keys []byte) bool {
+	// we need to write out a temp file for ssh-keygen to consume.
+	tf, err := ioutil.TempFile("", "update-ssh-keys-")
+	if err != nil {
+		panicf("unable to create temporary file: %v", err)
+	}
+	defer func() {
```

Comment:
Why wrap this in a func? Two defer lines would work just as well and in my eye looks cleaner.